lightning/src/ln/channel.rs

```diff
@@ -8027,10 +7977,41 @@ where
 		}
 	}
 
+	pub fn funding_transaction_signed(
+		&mut self, witnesses: Vec<Witness>,
+	) -> Result<(Option<msgs::TxSignatures>, Option<Transaction>), APIError> {
+		let (funding_tx_opt, tx_signatures_opt) = self
+			.interactive_tx_signing_session
+			.as_mut()
+			.ok_or_else(|| APIError::APIMisuseError {
+				err: format!(
+					"Channel {} not expecting funding signatures",
+					self.context.channel_id
+				),
+			})
+			.and_then(|signing_session| {
+				signing_session
+					.verify_interactive_tx_signatures(&self.context.secp_ctx, &witnesses)?;
+				signing_session
+					.provide_holder_witnesses(self.context.channel_id, witnesses)
+					.map_err(|err| APIError::APIMisuseError { err })
```

Comment:
Somewhat indifferent on this, but `verify_interactive_tx_signatures` could return a `String` error and be called by `provide_holder_witnesses`. That way it's impossible to pass invalid witnesses to `provide_holder_witnesses` without it erroring.